dfastmi/io/map_file.py

```diff
@@ -0,0 +1,347 @@
+
+                data = var[...]
+
+        return data
+
+    @property
+    def mesh2d_name(self) -> str:
+        """Get the name of the mesh2d variable.
+
+        Returns
+        -------
+        str
+            String containing the name of the mesh2d variable.
+        """
+        if not self._mesh2d_name:
+            with nc.Dataset(self._map_file) as rootgrp:
+                mesh2d = MapFile._get_mesh2d_variable(rootgrp)
+                self._mesh2d_name = mesh2d.name    
+ 
+        return self._mesh2d_name
+
+    @property
+    def face_dimension_name(self) -> str:
+        """Get the name of the face dimension.
+
+        Returns
+        -------
+        str
+            String containing the name of the face dimension.
+        """
+        if not self._face_dimension_name:
+            with nc.Dataset(self._map_file) as rootgrp:
+                mesh2d = rootgrp.variables[self.mesh2d_name]
+                facenodeconnect_varname = mesh2d.face_node_connectivity
+                fnc = rootgrp.get_variables_by_attributes(name=facenodeconnect_varname)[0]
+                self._face_dimension_name = fnc.dimensions[0]
+            
+        return self._face_dimension_name
+    
+    @staticmethod
+    def _get_mesh2d_variable(dataset: nc.Dataset) -> nc.Variable:
+        mesh2d = dataset.get_variables_by_attributes(
+            cf_role="mesh_topology", topology_dimension=2
+        )
+        if len(mesh2d) != 1:
+            raise ValueError(
+                "Currently only one 2D mesh supported ... this file contains {} 2D meshes.".format(
+                    len(mesh2d)
+                )
+            )
+        
+        return mesh2d[0]
+
+
+    @staticmethod
+    def _copy_var(source_dataset: nc.Dataset, var_name: str, target_dataset: nc.Dataset) -> None:
+        """
+        Copy a single variable from one netCDF file to another.
+
+        Copy a single netCDF variable including all attributes from source file to
+        destination file. Create dimensions as necessary.
+
+        Arguments
+        ---------
+        source_dataset : netCDF4.Dataset
+            Dataset object representing the source file.
+        var_name : str
+            Name of the netCDF variable to be copied from source to destination.
+        target_dataset : netCDF4.Dataset
+            Dataset object representing the destination file.
+        """
+        # locate the variable to be copied
+        variable = source_dataset.variables[var_name]
+
+        # copy dimensions
+        for dim_name in variable.dimensions:
+            dimension = source_dataset.dimensions[dim_name]
+            if dim_name not in target_dataset.dimensions.keys():
+                target_dataset.createDimension(
+                    dim_name, (len(dimension) if not dimension.isunlimited() else None)
+                )
+
+        # copy variable
+        variable_copy = target_dataset.createVariable(var_name, variable.datatype, variable.dimensions)
```

Comment:
Maybe move to a method called copy_variable?

Reply:
Done